android/src/main/java/com/it_nomads/fluttersecurestorage/FlutterSecureStoragePlugin.java

```diff
@@ -52,6 +55,23 @@ public void initInstance(BinaryMessenger messenger, Context context) {
       }
     }
 
+    /**
+     * This must be run in a separate Thread from an async method to avoid hanging UI thread on
+     * live devices in release mode.
+     * The most convenient place for that appears to be onMethodCall().
+     */
+    private void ensureInitStorageCipher() {
+        if(storageCipher == null) {
```

Comment:
it seems possible to bump into race condition here. Any sync primitives?

Reply:
I considered that. As written the code is not thread safe, but the channel methods are all run on the main UI thread in a serialized fashion (hence the UI blocking issue) so I don't *THINK* thread safety is an issue. Perhaps it would be better to add `synchronized` to the `ensureInitStorageCipher()` method signature to be 100% sure though.